actor/v7action/service_instance_sharing.go

```diff
@@ -18,9 +18,48 @@ func (actor Actor) ShareServiceInstanceToSpaceAndOrg(
 ) (Warnings, error) {
 	var serviceInstance resources.ServiceInstance
 	var shareSpace resources.Space
-	var shareToOrgGUID = targetedOrgGUID
 
 	return handleServiceInstanceErrors(railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			var actorWarnings Warnings
+			serviceInstance, shareSpace, actorWarnings, err = actor.validateSharingDetails(serviceInstanceName, targetedSpaceGUID, targetedOrgGUID, sharedToDetails)
+			warnings = ccv3.Warnings(actorWarnings)
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			_, warnings, err = actor.CloudControllerClient.ShareServiceInstanceToSpaces(serviceInstance.GUID, []string{shareSpace.GUID})
+			return
+		},
+	))
+
+	return Warnings{}, nil
+}
+
+func (actor Actor) UnshareServiceInstanceFromSpaceAndOrg(
+	serviceInstanceName, targetedSpaceGUID, targetedOrgGUID string,
+	sharedToDetails ServiceInstanceSharingParams,
+) (Warnings, error) {
+	return handleServiceInstanceErrors(railway.Sequentially(
```

Comment:
I think using the `handleServiceInstanceErrors` is redundant since it is being called on `validateSharingDetails`

This can be updated to be 
```
warnings, err := railway.Sequentially()
return Warnings(warnings), err
```

Reply:
the same for `ShareServiceInstanceToSpaceAndOrg `

Reply:
very good point